src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -2440,6 +2439,22 @@ void ApplicationManagerImpl::HeadUnitReset(
   }
 }
 
+void ApplicationManagerImpl::ClearUserStorage() {
+  const ApplicationSet& accessor = applications().GetData();
+
+  ApplicationSetConstIt it_app_list = accessor.begin();
+  ApplicationSetConstIt it_app_list_end = accessor.end();
+  const std::string storage_folder = get_settings().app_storage_folder();
+  for (; it_app_list != it_app_list_end; ++it_app_list) {
+    const std::string directory_path =
+        storage_folder + "/" + (*it_app_list).get()->folder_name();
+    file_system::RemoveDirectory(directory_path);
+    const std::string icon_path = get_settings().app_icons_folder() + "/" +
```

Comment:
@VVeremjova 
Could you, please create one more variable before entering the loop that contain get_settings().app_icons_folder() + "/" and concatenate it with the application dependent part in the loop body, because currently get_settings().app_icons_folder() is called for each application. You can also concatenate "/" to storage_folder variable.